src/entry_points/gentest.py

```diff
@@ -0,0 +1,303 @@
+    tr_str += pad + "v=" + str(tr.transaction.v) + ",\n"
+    tr_str += pad + "r=" + str(tr.transaction.r) + ",\n"
+    tr_str += pad + "s=" + str(tr.transaction.s) + ",\n"
+
+    test = test.replace("$TR", tr_str)
+    return test
+
+
+class PyspecConfig:
+    """
+    Main class to manage Pyspec config
+    """
+
+    @dataclass
+    class RemoteNode:
+        """
+        Remote node structure
+        """
+
+        name: str
+        node_url: str
+        client_id: str
+        secret: str
+
+    remote_nodes: List["PyspecConfig.RemoteNode"]
+
+    def __init__(self, config_path: str):
+        """
+        Initialize pyspec config from file
+        """
+        with open(config_path, "r") as file:
+            data = json.load(file)
+            self.remote_nodes = [self._json_to_remote_node(node) for node in data["remote_nodes"]]
+
+    def _json_to_remote_node(self, d):
+        return PyspecConfig.RemoteNode(
+            name=d["name"],
+            node_url=d["node_url"],
+            client_id=d["client_id"],
+            secret=d["secret"],
+        )
+
+
+class RequestManager:
+    """
+    Interface for the RPC interaction with remote node
+    """
+
+    @dataclass()
+    class RemoteTransaction:
+        """
+        Remote transaction structure
+        """
+
+        block_number: str
+        tr_hash: str
+        transaction: Transaction
+
+    node_url: str
+    headers: dict[str, str]
+
+    def __init__(self, node_config: PyspecConfig.RemoteNode):
+        """
+        Initialize the RequestManager with specific client config.
+        """
+        self.node_url = node_config.node_url
+        self.headers = {
+            "CF-Access-Client-Id": node_config.client_id,
+            "CF-Access-Client-Secret": node_config.secret,
+            "Content-Type": "application/json",
+        }
+
+    def eth_get_transaction_by_hash(self, transaction_hash: str) -> RemoteTransaction:
+        """
+        Get transaction data.
+        """
+        data = {
+            "jsonrpc": "2.0",
+            "method": "eth_getTransactionByHash",
+            "params": [f"{transaction_hash}"],
+            "id": 1,
+        }
+        response = requests.post(self.node_url, headers=self.headers, data=json.dumps(data))
+        res = response.json().get("result", None)
+
+        return RequestManager.RemoteTransaction(
+            block_number=res["blockNumber"],
+            tr_hash=res["hash"],
+            transaction=Transaction(
+                ty=int(res["type"], 16),
+                gas_limit=int(res["gas"], 16),
+                gas_price=int(res["gasPrice"], 16),
+                data=res["input"],
+                nonce=int(res["nonce"], 16),
+                sender=res["from"],
+                to=res["to"],
+                value=int(res["value"], 16),
+                v=int(res["v"], 16),
+                r=int(res["r"], 16),
+                s=int(res["s"], 16),
+            ),
```

Comment:
This one is a bit tricky, since it will only work with type-0 and type-1 transactions.

We could use a mapper maybe:
```python
transaction_field_mapper = {
  "type": "ty",
  "gas": "gas_limit",
  ...
}
```
To rename the fields in `res` and then simply pass `res` to the transaction constructor as `Transaction(**res)` (after removing `blockNumber` and `hash`).